### ISSUE TITLE
[Bug] Handling of `_FIVETRAN_DELETED`

### ISSUE THREAD

**Issue by raphaelvarieras** [fivetran/dbt_zuora_source#7]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the issue

After reviewing this package, I couldn't help but notice the absence of any logic in relations to the `_FIVETRAN_DELETED` tag in the source table. I don't see it in `src_zuora.yml`, and it doesn't seem to be something that can be added as a pass-through column either.

Aren't all the metrics from the package going to be wrong if it takes into account deleted accounts, deleted subscriptions, deleter invoices, etc?

### Relevant error log or model output

```shell
N/A
```

### Expected behavior

I would expect all the staging model to use this pattern:
```
with base as (

    select * 
    from {{ ref('stg_zuora__{entity}_tmp') }} where _fivetran_deleted is distinct from 'true'
),
```

### dbt Project configurations

N/A

### Package versions

  - package: fivetran/zuora # https://github.com/fivetran/dbt_zuora
    version: 0.1.0

### What database are you using dbt with?

snowflake

### dbt Version

1.5

### Are you willing to open a PR to help address this issue?

- [X] Yes.
- [ ] Yes, but I will need assistance and will schedule time during our [office hours](https://calendly.com/fivetran-solutions-team/fivetran-solutions-team-office-hours) for guidance
- [ ] No.

**Comment by fivetran-joemarkiewicz** [comment]
Hi @raphaelvarieras thanks for raising this issue. You are completely correct that we should handle the removal of the `_FIVETRAN_DELETED` field in the staging models to ensure accurate metrics in our downstream models.

In order to account for this we will need to do the following:
- Add the `_fivetran_deleted` field to the respective `macros/` files for each model
- Add a where statement to each model to filter out any deleted records directly in the staging model. We have handled this in the past with the [following logic](https://github.com/fivetran/dbt_salesforce_source/blob/main/models/stg_salesforce__event.sql#L62).
- Add the `_fivetran_deleted` field to our src.yml documentation

I see you are open to contributing a PR! If you are interested the I would follow the above steps and we can include those changes with your PR #6 so we may batch these updates together. Let me know if you are interested. If not, we can pick this update up in our upcoming sprint (starting mid next week).

**Comment by raphaelvarieras** [comment]
Ok, let me look into it, seems easy enough...

**Comment by fivetran-joemarkiewicz** [comment]
Sounds great! Let me know if you run into any issues or questions. I'm also happy to help at any point or take over the update if other priorities come your way 😄

**Comment by fivetran-joemarkiewicz** [comment]
Great, I will take a look tomorrow morning and we can see if it is good to fold in from there! I agree that `_fivetran_deleted` should not be included in the query results and instead be filtered out so I am on board with that approach. I will share any additional notes in the PR.

**Comment by fivetran-joemarkiewicz** [comment]
@raphaelvarieras thank you again for identifying and contributing these changes to the package! I have since merged these changes and cut a new v0.1.2 release of the zuora package. You should see these changes live on the dbt hub at the top of the hour. 

Thank you again and please feel free to comment if the issue still persists on your end.
